locksmith/src/models/sequelize.ts

```diff
@@ -1,9 +1,43 @@
-import { Sequelize } from 'sequelize'
+import { Sequelize, Utils, DataTypes } from 'sequelize'
 
 import logger from '../logger'
 import config from '../config/config'
 
 logger.info(`Connecting to database`)
 
-// We assume config from an object of {username, password, database, host, dialect}
+type LocksmithDataTypesType = typeof DataTypes & { NETWORK_ID: any }
+
+/**
+ * We need a custom type for networks.
+ * The chain/network id can be larger than the max INTEGER value in postgres, so we need to use BIGINT...
+ * But Sequelize, by default, treats this as BigInt and strings... which is not right because our app assumes
+ * these are numbers.
+ *
+ */
+// @ts-expect-error Base constructors must all have the same return type.
+class NETWORK_ID extends DataTypes.BIGINT {
+  // Optional: parser for values received from the database
+  static parse(value: string): number {
+    return Number.parseInt(value)
+  }
+}
+
```

Comment:
Do we also need to change validate, _sanitize, and stringify to only accept numbers and then convert to string before passing?

Reply:
Technically things will break when the `value` is for a number that's larger than JS's Number limit. For the rest I am ok to rely on `BIGINT`...

Reply:
Actually let me throw here when the number is too large.